helix-term/src/config.rs

```diff
@@ -1,35 +1,33 @@
 use anyhow::{Error, Result};
-use std::{collections::HashMap, str::FromStr};
+use std::collections::HashMap;
 
 use serde::{de::Error as SerdeError, Deserialize, Serialize};
 
 use crate::keymap::{parse_keymaps, Keymaps};
 
-pub struct GlobalConfig {
+#[derive(Default)]
+pub struct Config {
     pub theme: Option<String>,
-    pub lsp_progress: bool,
+    pub lsp: LspConfig,
+    pub keymaps: Keymaps,
 }
 
-impl Default for GlobalConfig {
-    fn default() -> Self {
-        Self {
-            lsp_progress: true,
-            theme: None,
-        }
-    }
+#[derive(Default)]
+pub struct LspConfig {
+    pub display_messages: bool,
```

Comment:
Drop LspTomlConfig, derive Serialize/Deserialize here and mark the key with #[serde(default)]

Reply:
https://serde.rs/attr-default.html

Reply:
Done.